bluemira/equilibria/plasma.py

```diff
@@ -0,0 +1,257 @@
+# bluemira is an integrated inter-disciplinary design tool for future fusion
+# reactors. It incorporates several modules, some of which rely on other
+# codes, to carry out a range of typical conceptual fusion reactor design
+# activities.
+#
+# Copyright (C) 2022 M. Coleman, J. Cook, F. Franza, I.A. Maione, S. McIntosh, J. Morris,
+#                    D. Short
+#
+# bluemira is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# bluemira is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with bluemira; if not, see <https://www.gnu.org/licenses/>.
+
+"""
+Representation of the plasma
+"""
+import numpy as np
+from scipy.interpolate import RectBivariateSpline
+
+from bluemira.equilibria.constants import J_TOR_MIN
+from bluemira.equilibria.error import EquilibriaError
+from bluemira.equilibria.plotting import PlasmaCoilPlotter
+from bluemira.magnetostatics.greens import greens_Bx, greens_Bz, greens_psi
+
+
+def treat_xz_array(func):
+    """
+    Decorator for handling array calls to PlasmaCoil methods.
+    """
+
+    def wrapper(self, x=None, z=None):
+        if x is None or z is None:
+            if z is None and x is None:
+                return func(self, x, z)
+            else:
+                raise EquilibriaError("Only one of x and z specified.")
+
+        x = np.array(x)
+        z = np.array(z)
+        if x.shape != z.shape:
+            raise EquilibriaError("x and z arrays of different dimension.")
+
+        values = np.zeros(x.shape)
+        values = values.flatten()
+
+        for i, (xx, zz) in enumerate(zip(x.flatten(), z.flatten())):
```

Comment:
https://stackoverflow.com/questions/30523123/when-to-use-flat-flatiter-or-flatten also ravel is an option (there are defintiely too many options...)
```suggestion
        for i, (xx, zz) in enumerate(zip(x.flat, z.flat)):
```